R/collect.R

```diff
@@ -60,9 +84,217 @@ collect_predictions <- function(x) {
+#  mode is used to summarize.
+
+numeric_summarize <- function(x, p, wider = TRUE) {
+  pred_cols <- grep("^\\.pred", names(x), value = TRUE)
+  nms <- names(x)
+  group_cols <- nms[!(nms %in% pred_cols)]
+  x <-
+    x %>%
+    tidyr::pivot_longer(
+      cols = c(dplyr::one_of(pred_cols)),
+      names_to = ".column",
+      values_to = ".value"
+    ) %>%
+    dplyr::group_by(!!!rlang::syms(group_cols), .column) %>%
+    dplyr::summarize(.value = mean(.value, na.rm = TRUE))
+
+  if (wider) {
+    x <-
+      x %>%
+      dplyr::ungroup() %>%
+      tidyr::pivot_wider(
+        id_cols = c(dplyr::one_of(group_cols)),
+        names_from = ".column",
+        values_from = ".value"
+      )
+  }
+  x
+}
+
+prob_summarize <- function(x, p, wider = TRUE) {
+  pred_cols <- grep("^\\.pred", names(x), value = TRUE)
+  nms <- names(x)
+  group_cols <- nms[!(nms %in% pred_cols)]
+  x <-
+    x %>%
+    tidyr::pivot_longer(
+      cols = c(dplyr::one_of(pred_cols)),
+      names_to = ".column",
+      values_to = ".value"
+    ) %>%
+    dplyr::group_by(!!!rlang::syms(group_cols), .column) %>%
+    dplyr::summarize(.value = mean(.value, na.rm = TRUE))
+
+  avg_group_cols <- group_cols[group_cols != ".column"]
+  x <-
+    x %>%
+    dplyr::full_join(
+      x %>% dplyr::summarize(.totals = sum(.value)),
+      by = avg_group_cols
+    ) %>%
+    dplyr::mutate(.value = .value/.totals) %>%
+    dplyr::select(-.totals)
+
+  if (wider) {
+    x <-
+      x %>%
+      dplyr::ungroup() %>%
+      tidyr::pivot_wider(
+        id_cols = c(dplyr::one_of(group_cols)),
+        names_from = ".column",
+        values_from = ".value"
+      )
+  }
+  x
+}
+
+prob_and_class_summarize <- function(x, p) {
+  x$.pred_class <- NULL
+  x <- prob_summarize(x, p, wider = FALSE)
+  nms <- names(x)
+  group_cols <- nms[!(nms %in% c(".column", ".value"))]
+  outcome_col <- group_cols[!(group_cols %in% c(p, ".row", ".iter"))]
+  lvl <- levels(x[[outcome_col]])
+  ord <- is.ordered(x[[outcome_col]])
+  y <-
+    x %>%
+    dplyr::arrange(dplyr::desc(.value), .by_group = TRUE) %>%
+    dplyr::slice(1) %>%
+    dplyr::mutate(
+      .value = gsub("\\.pred_", "", .column),
+      .value = factor(.value, levels = lvl, ordered = ord),
+      .column = ".pred_class") %>%
+    dplyr::ungroup() %>%
+    tidyr::pivot_wider(
+      id_cols = c(dplyr::one_of(group_cols)),
+      names_from = ".column",
+      values_from = ".value"
+    )
+  x <-
+    x %>%
+    dplyr::ungroup() %>%
+    tidyr::pivot_wider(
+      id_cols = c(dplyr::one_of(group_cols)),
+      names_from = ".column",
+      values_from = ".value"
+    ) %>%
+    dplyr::full_join(y, by = group_cols)
+  x
+}
+
+class_summarize <- function(x, p) {
+  pred_cols <- grep("^\\.pred", names(x), value = TRUE)
+  nms <- names(x)
+  group_cols <- nms[!(nms %in% pred_cols)]
+  outcome_col <- group_cols[!(group_cols %in% c(p, ".row", ".iter"))]
+  lvl <- levels(x[[outcome_col]])
+  ord <- is.ordered(x[[outcome_col]])
```

Comment:
Do you need these?

Reply:
Yes since I'll be creating a new factor later and I want it to match the observed classes.

Reply:
nvmd; I was looking at the wrong function; yes, those are also vestigial.